cEP-0010.md

```diff
@@ -0,0 +1,136 @@
+in [cEP](https://github.com/coala/cEPs/blob/master/cEP-0005.md). Our concern
+here is to enumerate and implement all the `aspects` of code the bears offered
+by coala are capable of analyzing and, and annotating bears' results with them.
+
+## The aspests
+
+The bears offered by coala are capable of several aspects of code, aspects will
+be implemented in a tree-like form with root the [Root aspects](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/root.py).
+Here is the list of aspects to be implemented along with their corresponding
+`subaspects` and tastes or settings(which are the leaves of the tree.
+
+
+
+  Root
+  |--- Formatting
+  |--- |--- UseSpaces
+  |--- |--- LineLength
+  |--- |--- |--- max_line_length
+  |--- |--- |--- Ignore
+  |--- |--- |--- |--- links
+  |--- |--- |--- |--- line
+  |--- |--- Indentation
+  |--- |--- |--- size
+  |--- |--- ParenthesesFormatting
+  |--- |--- BracesFormatting
+  |--- Smell
+  |--- |--- ClassSmell
+  |--- |--- |--- ClassLength
+  |--- |--- |--- |--- max_class_length
+  |--- |--- |--- |--- min_class_length
+  |--- |--- |--- feature_envy
+  |--- |--- MethodSmell
+  |--- |--- |--- max_method_length
+  |--- |--- |--- max_parameter
+  |--- |--- Naming
+  |--- |--- |--- naming_convention
+  |--- |--- |--- max_indentifier_name_length
+  |--- |--- |--- min_indentifier_name_length
+  |--- Security
+  |--- |--- buffer_overflow
+  |--- Redundancy
+  |--- |--- Clone
+  |--- |--- UnusedImport
+  |--- |--- UnreachableCode
+  |--- |--- UnusedFunction
+  |--- |--- UnreachableStatement
+  |--- |--- UnusedVariable
+  |--- |--- UnusedParameter
+  |--- |--- UnusedLocalVariable
+  |--- |--- UnusedGlobalVariable
+  |--- Complexity
+  |--- |--- cyclomatic_complexity
+  |--- Spelling
+  |--- |--- aspectsYEAH
+  |--- |--- coala
+  |--- Smell
+  |--- Undefined elements
+  |--- |--- UndefinedVariable
+  |--- |--- UndefinedParameter
+  |--- |--- UndefinedFunction
+  |--- Documentation
+  |--- Grammar
+  |--- |--- Language
+  |--- Metadata
+  |--- |--- CommitMessage
+  |--- |--- | --- Shortlog
+  |--- |--- | --- |--- Emptiness
+  |--- |--- | --- |--- ColonExistence
+  |--- |--- | --- |--- TrailingPeriod
+  |--- |--- | --- |--- Tense
+  |--- |--- | --- |--- Length
+  |--- |--- | --- |--- FirstCharacter
+  |--- |--- | --- Body
+  |--- |--- | --- |--- Existence
+  |--- |--- | --- |--- Length
+
+
+## Implementation
+
+As stated previously the aspects will be implemented in a tree like form,
+with the Root aspects at the root of the tree. Some classes, metaclasses
+and some few methods were define so far to make that possible including :
+
+* [the aspectbase metaclass](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/base.py)
+* [the Documentation class for documenting aspects](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/docs.py)
+* [the aspectclass](https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/meta.py)
```

Comment:
`aspectclass` is the actual metaclass :) So you could name the link `[the meta aspectclass](...)`